libraries/core/classes/PipelineManager/index.js

```diff
@@ -242,56 +249,57 @@ class PipelineManager {
       serial,
     }, logColor);
 
+    // Cleanup.
+    event.removeCallback(callbackName, request.callback);
     clearTimeout(entry.timer);
+    this.removeRequestFromPiplineSequence(serial);
     this.requests.delete(serial);
+
+    // Take care about requests that where deferred since depencensies where running.
+    this.handleDeferredRequests();
   }
 
   /**
    * Handles the results sequentially.
    */
-  handleResultSequence = () => {
-    const sequence = pipelineSequence.get();
-
-    /* eslint-disable no-restricted-syntax */
-    for (const ser of sequence) {
-      const entry = this.requests.get(ser);
-
-      if (!entry.request.output) {
+  handleResultSequence() {
+    // Create a copy of the sequence, to avoid side effects when entries are removed.
+    const [...sequence] = pipelineSequence.get();
+
+    for (let i = 0; i < sequence.length; i += 1) {
+      const serial = sequence[i];
+      const entry = this.requests.get(serial);
+
+      if (!entry) {
+        // Remove sequence entries without request.
+        this.removeRequestFromPiplineSequence(serial);
+      } else if (!entry.finished) {
+        // Stop sequence procession at the first not finished request.
         break;
+      } else {
```

Comment:
No need to else here if first if statement would contain `break`

Reply:
Since after the removal of a sequence entry the rest of the entries should still be processed, the alternatives would be `continue` - which the linter doesn't like - or a `return.`

Reply:
OK